stages/test/test_mkfs_xfs.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+
+import os.path
+import re
+import subprocess
+from unittest import mock
+
+import pytest
+
+from osbuild import testutil
+from osbuild.testutil import has_executable
+
+STAGE_NAME = "org.osbuild.mkfs.xfs"
+
+
+# Prepare dataset containing good and bad API call parameters
+@pytest.mark.parametrize("test_data, expected_err", [
+    # Bad API parameters
+    ({}, "'uuid' is a required property"),
+    ({"uuid": 123}, "123 is not of type 'string'"),
+    ({"uuid": "text", "label": 1234}, "1234 is not of type 'string'"),
+    ({"uuid": "text", "label": "12345678901234567"}, "is too long"),
+    # Good API parameters
+    # mkfs requires valid UUID, but API will accept any string right now.
+    # Bellow will fail when API will be more restricted in the future.
+    ({"uuid": "text"}, ""),
+    ({"uuid": "text", "label": "123456789012"}, ""),
+])
+# This test validates only API calls using correct and incorrect queries
+def test_schema_validation_mkfs_xfs(stage_schema, test_data, expected_err):
+    test_input = {
+        "type": STAGE_NAME,
+        "devices": {
+            "device": {
+                "path": "some-path",
+            },
+        },
+        "options": {
+        }
+    }
+    test_input["options"].update(test_data)
+    res = stage_schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+# This test creates dummy filesystem using predefined parameters and verifies
+# that end result has those parameters set
+@pytest.mark.skipif(not has_executable("mkfs.xfs"), reason="need mkfs.xfs")
+def test_mkfs_fat_integration(tmp_path, stage_module):
+    fake_disk_path = tmp_path / "fake.img"
+    with fake_disk_path.open("w"):
+        os.truncate(os.fspath(fake_disk_path), 300 * 1024 * 1024)
+    devices = {
+        "device": {
+            "path": fake_disk_path,
+        },
+    }
+    fake_uuid = "6d35d58f-db32-489b-99a0-08fb10e9aa29"
```

Comment:
(nitpick) I would generate a real random uuid here. I don't think in this case it matters but it seems good practice. We got recently bitten by a non-random uuid element in https://github.com/osbuild/osbuild/pull/1641. But here it's different AFAICT so not strictly needed.

Reply:
True. My point for using copy-paste uid here is to easier spot it in the eventual fail logs. It will be always the same, so user will not have to check what was generated first and then look for it in case of failure. I think that risk of overlaping with some real UID is minimal.

Reply:
Sure, fine to leave as is if you feel strongly about it. Fwiw, I'm not worried about overlap with real UUIDs but more about when things are run in parallel. The PR I linked to above was caused by a hardcoded UUID and then two variants of the test ran in parallel and because the system assumed the uuid is unique (which it usually is) things started to fail and it was hard to debug. So my suggestion would still be to do:
```python
diff --git a/stages/test/test_mkfs_xfs.py b/stages/test/test_mkfs_xfs.py
index 3be7b45f..635a7fe1 100644
--- a/stages/test/test_mkfs_xfs.py
+++ b/stages/test/test_mkfs_xfs.py
@@ -3,6 +3,7 @@
 import os.path
 import re
 import subprocess
+import uuid
 from unittest import mock
 
 import pytest
@@ -60,7 +61,7 @@ def test_mkfs_xfs_integration(tmp_path, stage_module):
             "path": fake_disk_path,
         },
     }
-    fake_uuid = "6d35d58f-db32-489b-99a0-08fb10e9aa29"
+    fake_uuid = uuid.uuid4()
     fake_label = "osbuild"
     options = {
         "uuid": fake_uuid,
```
even if it does not matter today. But code evolves and gets copied/reused and also it's (IMO) a good pattern to not hardcode things like UUIDs (we could try to make them predicatable with a random seed or something but that seems overkill in this case).

But really up to you, this is very minor, I'm okay either way.

Reply:
Done:)